spark/src/main/scala/ai/chronon/spark/Analyzer.scala

```diff
@@ -407,8 +407,25 @@ class Analyzer(tableUtils: TableUtils,
         )
       }
     }
+    val aggMetadata: ListBuffer[AggregationMetadata] = if (joinConf.hasDerivations) {
+      val keyAndPartitionFields =
+        leftDf.schema.fields ++ Seq(org.apache.spark.sql.types.StructField(tableUtils.partitionColumn, StringType))
+      val sparkSchema = {
+        val schema: Seq[(String, DataType)] = leftSchema.toSeq ++ rightSchema.toSeq
+        StructType(SparkConversions.fromChrononSchema(schema))
+      }
+      val dummyOutputDf = tableUtils.sparkSession
+        .createDataFrame(tableUtils.sparkSession.sparkContext.parallelize(immutable.Seq[Row]()), sparkSchema)
+      val finalOutputColumns = joinConf.derivationsScala.finalOutputColumn(dummyOutputDf.columns).toSeq
+      val derivedDummyOutputDf = dummyOutputDf.select(finalOutputColumns: _*)
+      val columns = SparkConversions.toChrononSchema(
+        StructType(derivedDummyOutputDf.schema.filterNot(f => keyAndPartitionFields.map(_.name).contains(f.name))))
```

Comment:
The aggMetadata are actually features, so the purpose of this line is to exclude the keys from the features list. 
But this may not apply to the case where the users rename an external feature to a key name, if that happens the key(which is actually an external feature) will be filtered out here 🤔